janusgraph-es/src/main/java/org/janusgraph/diskstorage/es/ElasticSearchIndex.java

```diff
@@ -259,16 +258,32 @@
      */
     public static final double DEFAULT_GEO_DIST_ERROR_PCT = 0.025;
 
+    private static final String PARAMETERIZED_DELETION_SCRIPT = String.join(" ", "",
+            "for (field in params.fields) {",
+            "    if (field.cardinality == 'SINGLE') {",
+            "        ctx._source.remove(field.name);",
+            "    } else if (ctx._source.containsKey(field.name)) {",
+            "        def fieldIndex = ctx._source[field.name].indexOf(field.value);",
+            "        if (fieldIndex >= 0 && fieldIndex < ctx._source[field.name].size()) {",
+            "            ctx._source[field.name].remove(fieldIndex);",
+            "        }",
+            "    }",
+            "}");
+
+    private static final String PARAMETERIZED_ADDITION_SCRIPT = String.join(" ", "",
+            "for (field in params.fields) {",
+            "    if (ctx._source[field.name] == null) {",
+            "        ctx._source[field.name] = [];",
+            "    }",
+            "    if (field.cardinality == 'SET' && ctx._source[field.name].indexOf(field.value) != -1) {",
+            "        break;",
```

Comment:
Can `params.fields` contain fields with different cardinality?
If so, I am not sure it is right to break the loop here. Maybe you thought about `continue;`?
I think we can even invert it to be like:
```
if (field.cardinality != 'SET' || ctx._source[field.name].indexOf(field.value) == -1) {
    ctx._source[field.name].add(field.value);
}
```

Reply:
OMG  I meant `continue` there. Good catch.